src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,108 @@
+
+import java.nio.file.Paths;
+
+/**
+ * Represents the highest level code responsible for Duke's operations.
+ */
+public class Duke {
+
+
+    public static void main(String[] args) {
+
+        TaskList taskList = new TaskList();
+        Scanner sc = new Scanner(System.in);
+        Storage storage = new Storage(Paths.get("data", "duke.txt"), Paths.get("data"));
+
+        taskList = new TaskList(storage.readFromFile());
+
+        Ui.showInitUi();
+
+        while (sc.hasNextLine()) {
+
+            try {
+                DukeCommand dukeCommand = Parser.parseCommand(sc.nextLine());
+                if (dukeCommand.getCommand() == Command.BYE) {
+                    Ui.showExitUi();
+                    sc.close();
+                    return;
+                } else if (dukeCommand.getCommand() == Command.DELETE) {
+                    Integer index = Integer.parseInt(dukeCommand.getDetails()) - 1;
+
+                    if (index >= taskList.getTasks().size()) {
+                        throw new DukeException("No such task in the list");
+                    }
+
+                    Task removedTask = taskList.delete(Integer.parseInt(dukeCommand.getDetails()) - 1);
+                    storage.writeToFile(taskList);
+                    Ui.showSuccessfulDelete(taskList.getTasks().size(), removedTask);
+
+
+                } else if (dukeCommand.getCommand() == Command.LIST) {
+                    Ui.showList(taskList);
+
+                } else if (dukeCommand.getCommand() == Command.DONE) {
+                    Integer index = Integer.parseInt(dukeCommand.getDetails()) - 1;
+
+                    if (index >= taskList.getTasks().size()) {
+                        throw new DukeException("No such task in the list");
+                    }
+
+                    taskList.markAsDone(index);
+                    storage.writeToFile(taskList);
+                    Ui.showSuccessfulDone(taskList.get(index));
+
+
+                } else if (dukeCommand.getCommand() == Command.INVALID) {
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-()");
+
+                } else if (dukeCommand.getCommand() == Command.FIND) {
+                    if (dukeCommand.getDetails().length() == 0) {
+                        throw new DukeException("☹ OOPS!!! Search keyword cannot be empty");
+                    }
+
+                    ListIterator<Task> taskIter = taskList.getTasks().listIterator();
+                    String keyword = dukeCommand.getDetails();
+                    TaskList matchedTasks = new TaskList();
+
+                    while (taskIter.hasNext()) {
+                        Task curr = taskIter.next();
+
+                        if (curr.getDescription().contains(keyword)) {
+                            matchedTasks.add(curr);
+                        }
+                    }
+
+                    Ui.showMatchedTasks(matchedTasks);
+
+
+                } else {
```

Comment:
Could these blocks be simplified / abstracted out? ([Textbook section](https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent))